Source/Core/Request.js

```diff
@@ -0,0 +1,64 @@
+/*global define*/
+define([
+        'Core/defaultValue'
+    ], function(
+        defaultValue) {
+    "use strict";
+
+    /**
+     * Stores information for making a request using {@link RequestScheduler}.
+     *
+     * @exports Request
+     *
+     * @private
+     */
+    function Request(options) {
+        options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+
+        /**
+         * The URL to request.
+         */
+        this.url = options.url;
+
+        /**
+         * Extra parameters to send with the request. For example, HTTP headers or jsonp parameters.
+         */
+        this.parameters = options.parameters;
+
+        /**
+         * The actual function that makes the request.
+         */
+        this.requestFunction = options.requestFunction;
+
+        /**
+         * Type of request. Used for more fine-grained priority sorting.
+         */
+        this.requestType = options.requestType;
```

Comment:
This should probably be `type`, e.g., `request.type` instead of `request.requestType`.

Same comment for `requestFunction`, except we would have to use `func` (I think) so maybe just stay with `requestFunction`.